### PR TITLE
Timeslots

### PR DESCRIPTION
**Pull request by KiraBurova**
- [x] Used appropriate errors codes
- [x] Fixed logic condition
- [x] Added a test for an edge case 
- [x] Get rid of the loop in handler
- [x] Remove timeslots on passed date before inserting the new ones
- [x] Fix tests

### CODE REVIEW

File: timeslots/repository.go [hunk 1]
```diff
@@ -15,47 +18,88 @@ func NewRepository(db *sql.DB) *Repository {
...
 
-	// for i := 0; i < len(timeslots); i++ {
-	// add id, TimeFrom, TimeTo here
-	//}
+	bookingDayInUnix := timeslot.BookingDay.Unix()
 
-	// for _, timeslot := range timeslots {
-	// somehow construct SQL query here
-	// }
+	ts, err := r.getTimeslotByBookingDay(bookingDayInUnix)
```
Comment (KiraBurova) [inline timeslots/repository.go:54]: add check if err != nil && not no rows

File: timeslots/repository.go [hunk 2]
```diff
@@ -15,47 +18,88 @@ func NewRepository(db *sql.DB) *Repository {
...
 
-	timeInUnixFrom := timeslot.TimeFrom.Unix()
-	timeInUnixTo := timeslot.TimeTo.Unix()
+	if !errors.Is(err, sql.ErrNoRows) {
```
Comment (KiraBurova) [inline timeslots/repository.go:56]: instead of `sql.ErrNoRows` sql request count, if count > 0 -> there is a timeslot -> delete it

File: timeslots/timeslots.go [hunk 3]
```diff
@@ -33,7 +34,7 @@ func isTimePeriodValid(timeperiod TimePeriod) bool {
 	return timeperiod.From.Before(timeperiod.To)
 }
 
-func areTimePeriodsOverlapping(timeperiods []TimePeriod) bool {
+func areTimePeriodsNotOverlapping(timeperiods []TimePeriod) bool {
```
Comment (KiraBurova) [inline timeslots/timeslots.go:37]: rename back and fix function logic

**Review by rutaka-n — APPROVED** [review]
